include/onnxruntime/core/optimizer/graph_transformer.h

```diff
@@ -130,8 +163,8 @@ This is a rule-based Graph transformer that applies rules by performing top-down
 */
 class TopDownRuleBasedTransformer : public RuleBasedGraphTransformer {
  public:
-  TopDownRuleBasedTransformer(const std::string& name, const std::string& desc)
-      : RuleBasedGraphTransformer(name, desc) {}
+  TopDownRuleBasedTransformer(const std::string& name, const std::string& desc, const TransformerLevel level, std::vector<std::string> providers)
```

Comment:
To keep this design working well on transformer and not remove the extension for rewrite rules. Let's have rule transformer having level 0 now, which means it will always be run in every level, so that it can control/design levels for rules if needed later. Sounds good?